api/graphql/reservation_units/reservation_unit_types.py

```diff
@@ -169,12 +170,47 @@ class Meta:
         fields = ("url",)
         connection_class = TilavarausBaseConnection
 
-    def resolve_url(self, info):
+    def __init__(
+        self,
+        *args,
+        instance: ReservationUnit = None,
+        include_reservation_units: List[int] = None,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+        if instance:
+            self.instance = instance
+            self.uuid = instance.uuid
+            self.unit = instance.unit
+            self.include_reservation_units = include_reservation_units
+
+    def resolve_url(self, info, **kwargs):
         if can_manage_units(info.context.user, self.unit):
+            target_uuids = []
+            include_reservation_units = getattr(self, "include_reservation_units", None)
+
+            if include_reservation_units:
+                res_units_in_db = ReservationUnit.objects.filter(
+                    id__in=include_reservation_units
+                )
+
+                difference = set(include_reservation_units).difference(
+                    {res_unit.id for res_unit in res_units_in_db}
+                )
+
+                if difference:
+                    raise GraphQLError(
+                        "Wrong identifier for reservation unit in url generation."
+                    )
+                for res_unit in res_units_in_db.filter(unit=self.unit):
+                    target_uuids.append(res_unit.uuid)
+
```

Comment:
The updated lines look great!

You can do away with the loop here with the `values_list()` queryset function:

```python
target_uuids = res_units_in_db.filter(unit=self.unit).values_list("uuid", flat=True)
```